lib/src/diff/main.dart

```diff
@@ -296,21 +296,21 @@ List<Diff> diffBisect(String text1, String text2, double timeout,
   int k2end = 0;
   for (int d = 0; d < max_d; d++) {
     // Bail out if deadline is reached.
-    if ((new DateTime.now()).compareTo(deadline) == 1) {
+    if ((new DateTime.now()).compareTo(deadline!) == 1) {
       break;
     }
 
     // Walk the front path one step.
     for (int k1 = -d + k1start; k1 <= d - k1end; k1 += 2) {
       int k1_offset = v_offset + k1;
-      int x1;
-      if (k1 == -d || k1 != d && v1[k1_offset - 1] < v1[k1_offset + 1]) {
+      int? x1;
```

Comment:
x1 is definitly set in line 308 or 310. So you could assign a dummy value and avoid all the ! after it.
Like
int x1 = 0;